### PR TITLE
[#16157] Pinned list button is absent when user is located at the top…

### LINKED ISSUE #16157: Pinned list button is absent when user is located at the top of the chat history
Introduced by https://github.com/status-im/status-mobile/pull/15204 cc @alwx 

Causing pinned messages e2e tests to fail. 

**Steps:** 

1. Open any chat
2. Send some messages
3. Pin a message to generate pinned list
4. Scroll to the top of chat history 
5. See if pinned list button is availabe

**Actual result:** pinned list button is not shown. Unable to open pinned list as pinned list bar disappears when user located at the top of chat history. 

**Expected result:** pinned list button is displed as per design

https://www.figma.com/file/7KIYbhoqNGAIFonE0w9TDz/Messages-for-Mobile?type=design&node-id=7950-519665&t=XaUGsCIHDWIbcWct-0

Messages for Mobile – Figma 2023-06-02 11-59-38

### Additional Information

- Status version: nightly 
- Operating System: Android, iOS

### PR DESCRIPTION
**Pull request by flexsurfer**
fixes #16157

accessibility-label `action-button-pinned`

### CODE REVIEW

File: ios/Podfile.lock [hunk 1]
```diff
@@ -787,4 +787,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: a7c3cb360cf217ab90667d67deeab588677d540a
 
-COCOAPODS: 1.12.0
+COCOAPODS: 1.12.1
```
Comment (flexsurfer) [inline ios/Podfile.lock:790]: someone reverted this at some point

**Review by smohamedjavid — APPROVED** [review]

File: ios/Podfile.lock (hunk 1 above)
Comment (smohamedjavid) [reply ios/Podfile.lock:790]: I believe the  Cocoapods version inside Nix is `1.12.0`.

**Comment by siddarthkay** [comment]
@flexsurfer : Sorry, I'll review this PR tomorrow, I'm out of my bandwidth for today.

File: src/quo2/components/community/channel_actions.cljs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+     (when counter-value
+       [counter/counter {:type :secondary} counter-value])]
+    [text/text {:size :paragraph-1 :weight :medium} label]]])
+
+(defn channel-actions
+  [{:keys [style actions]}]
+  [rn/view {:style (merge {:flex-direction :row :flex 1} style)}
+   (for [action actions]
+     [:<>
+      [channel-action action]
```
Comment (Parveshdhull) [inline src/quo2/components/community/channel_actions.cljs:26]: should we add a key here?

File: src/status_im2/contexts/quo_preview/community/channel_actions.cljs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    [channel-actions/channel-actions
+     {:actions [{:label "Pinned Messages" :color :blue :icon :i/pin :counter-value 5}
+                {:label "Mute chat" :color :blue :icon :i/muted}]}]
+
+    [rn/view {:height 50}]
+    [channel-actions/channel-actions
+     {:actions [{:big? true :label "Pinned Messages" :color :blue :icon :i/pin :counter-value 5}
+                {:label "Mute chat" :color :blue :icon :i/muted}]}]
+
+
```
Comment (Parveshdhull) [inline src/status_im2/contexts/quo_preview/community/channel_actions.cljs:25]: empty line

**Review by Parveshdhull — APPROVED** [review]

**Comment by pavloburykh** [comment]
@flexsurfer thanx for the PR! Please take a look at the following issues

### ISSUE 1 Multiline pinned message preview appears out of pinned button border (Android)

Tested on Android 12, Samsung Galaxy A52

**Steps:** 

1. Pin multiline message (4+ lines)
2. Observe how message preview is displayed inside pinned button

**Actual result:** 

 
photo_2023-06-22 18 26 32

**Comment by pavloburykh** [comment]
### ISSUE 2 Wrong pinned icon color inside pinned button in dark mode until app re-login

This is minor issue. Might be not PR related but part of global issues related to dark mode enabling. 

**Steps:** 

1. Enable dark mode
2. Pay attention at pinned icon color inside pinned bottom

**Actual result:** icon is black right after changing mode to dark. After app re-login it becomes light. 

photo_2023-06-22 18 37 14

**Expected result:** icon is light

photo_2023-06-22 18 39 50

**Comment by flexsurfer** [comment]
thanks @pavloburykh fixed

**Comment by pavloburykh** [comment]
@flexsurfer thank you for the fixes!

Could you please address one more minor: 

### ISSUE 3 Three dots are not displayed at the end of long truncated text inside pinned button (IOS)

Reproducing om IOS 14.1, iPhone X

**Actual result:** 

photo_2023-06-23 14 05 00

**Expected result:** 

photo_2023-06-23 14 04 34

**Comment by flexsurfer** [comment]
hey @pavloburykh thanks, i don't think we can fix it, but if you want you can file a separate low prio issue, thanks

**Comment by Parveshdhull** [comment]
Small reminder @flexsurfer Please address above review comments before merging PR.

**Comment by pavloburykh** [comment]
> hey @pavloburykh thanks, i don't think we can fix it, but if you want you can file a separate low prio issue, thanks

got it, thanx. PR is ready for merge then.

**Comment by pavloburykh** [comment]
> Small reminder @flexsurfer Please address above review comments before merging PR.

@flexsurfer let me know if re-test is needed after addressing review comments.

**Pull request merged**